// route_handler/index.js
const MongoClient = require('mongodb').MongoClient;

/*conntct to DB*/
const mongoClient = new MongoClient("mongodb://localhost:27017", {useNewUrlParser: true, useUnifiedTopology: true});
mongoClient.connect(function(err, client){
	//connect
	const db = client.db("userdb");
	//select table
	const collection = db.collection("users");
	//create user
	let user = {
		name: "User2",
		age: "35",
		login: "user",
		password: "123123",
		val: "123"
	};

	//insert
	collection.insertOne(user, function(err, res){
		if (err) {
			console.log(err);
		}
		else {
			console.log(res.ops);
		}
		client.close();
	});

	//find
	collection.find({login: 'admin'}).toArray(function(err, res){
		if (err) {
			console.log(err);
		}
		else {
			console.log(res);
		}
		client.close();
	});
    //delete
	collection.deleteMany({login: 'user'}), (function(err, res){
		if (err) {
			console.log(err);
		}
		else {
			console.log(res);
		}
		client.close();
	});	

	//update
	collection.findOneAndUpdate(
		{name: 'User1'},
		{$set: {age:"51"}},
		{returnOriginal: false}
		, function(err, res){
		if (err) {
			console.log(err);
		}
		else {
			console.log(res);
		}
		client.close();
	});		
});


var route_handler = {
	index: function(){
		return "This is home page";
	},
	contacts: function(){
		return "Contacts page";
	}
};

module.exports = route_handler;
